core/src/lib.rs

```diff
@@ -390,11 +396,36 @@ pub fn flatten_ident(ident: &Ident) -> Ident {
 ///
 /// Used by [`export_tokens_internal`] and several other functions.
 pub fn export_tokens_macro_ident(ident: &Ident) -> Ident {
-    let ident = flatten_ident(&ident);
+    let ident = flatten_ident(ident);
     let ident_string = format!("__export_tokens_tt_{}", ident.to_token_stream().to_string());
     Ident::new(ident_string.as_str(), Span::call_site())
 }
 
+pub fn export_tokens_macro_path(item_path: &Path) -> Path {
+    let Some(last_seg) = item_path.segments.last() else { unreachable!("must have at least one segment") };
+    let mut leading_segs = item_path
+        .segments
+        .iter()
+        .cloned()
+        .map(|seg| seg.ident)
+        .collect::<Vec<_>>()[0..item_path.segments.len() - 1]
+        .to_vec();
+    let last_seg = export_tokens_macro_ident(&last_seg.ident);
+    leading_segs.push(last_seg);
+    parse_quote!(#(#leading_segs)::*)
```

Comment:
maybe this is slightly cleaner as we don't loose the path leading colon, what do you think?
```suggestion
    let mut macro_path = item_path.clone();
    let Some(syn::punctuated::Pair::End(last_seg)) = macro_path.segments.pop()
        else { unreachable!("Path must have at least one last segment") };
    let last_seg = export_tokens_macro_ident(&last_seg.ident);
    macro_path.segments.push(last_seg.into());
    macro_path
```